navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -136,44 +137,152 @@ private static void putLegInformation(ObjectNode legJson, ResponsePath path, int
-                for (Integer bearing : bearingsList) {
-                    bearingsrray.add(bearing);
+
+                List<Integer> mergedBearings = new ArrayList<>();
+                List<Boolean> mergedEntry = new ArrayList<>();
+
+                // We copy over all bearings and entries for all edges but for the Outgoing Edge
+                int mergedIn = -1;
+                for (int b = 0; b < bearings.size(); b++) {
+                    if (b == out) {
+                        continue;
+                    }
+                    mergedBearings.add( bearings.get(b));
+                    mergedEntry.add( entry.get(b));
+                    if ( b == in) {
+                        mergedIn = mergedBearings.size()-1;
+                    }
                 }
-                // in
-                if (intersectionValue.containsKey("in")) {
-                    intersection.put("in", (int) intersectionValue.get("in"));
+
+                Map<String, Object> nextIntersection = (Map<String, Object>) list.get(i+1).getValue();
+                List<Integer> nextBearings = (List<Integer>) nextIntersection.get("bearings");
+                Integer nextIn = (Integer) nextIntersection.get("in");
+                Integer nextOut = (Integer) nextIntersection.get("out");
+                List<Boolean> nextEntry = (List<Boolean>) nextIntersection.get("entries");
+
+                if ( nextBearings.get(nextIn) != 90) {
+                    // oops, it should have a bearing of 90
+                    continue;
                 }
-                // out
-                if (intersectionValue.containsKey("out")) {
-                    intersection.put("out", (int) intersectionValue.get("out"));
+                // We copy over all bearings and entries for all edges but for the Incoming
+                int mergedOut = -1;
+                for (int b = 0; b < nextBearings.size(); b++) {
+                    if ( b == nextIn) {
+                        continue;
+                    }
+                    mergedBearings.add( nextBearings.get(b));
+                    mergedEntry.add( nextEntry.get(b));
+                    if ( b == nextOut) {
+                        mergedOut = mergedBearings.size()-1;
+                    }
                 }
+
+                // remove the duplicate Intersection
+                list.remove(i+1);
+                Map<String, Object> mergedIntersection = new HashMap<>();
+                mergedIntersection.put( "in", mergedIn);
+                mergedIntersection.put( "out", mergedOut);
+                mergedIntersection.put( "bearings", mergedBearings);
+                mergedIntersection.put( "entries", mergedEntry);
+
+                PathDetail mergedPathDetail = new PathDetail( mergedIntersection);
+                mergedPathDetail.setFirst( list.get(i).getFirst());
+                // and replace the intersection with the merged one
+                list.set( i, mergedPathDetail);
```

Comment:
formatting seems wrong in this file? `( ` many extra spaces after the opening bracket, mostly in the marked section but also in other `if ( ` conditions

Reply:
Reformatted based on .editorconfig in the repositoriy